crates/bioauth-flow-rpc/src/errors/authenticate_v2.rs

```diff
@@ -0,0 +1,163 @@
+//! The `authenticate_v2` method error.
+
+use rpc_validator_key_logic::Error as ValidatorKeyError;
+
+use super::{api_error_code, shared};
+use crate::error_data::{self};
+
+/// The `authenticate_v2` method error kinds.
+#[derive(Debug)]
+pub struct Error(pub shared::FlowBaseError<robonode_client::AuthenticateError>);
+
+impl From<Error> for jsonrpsee::core::Error {
+    fn from(err: Error) -> Self {
+        match err.0 {
+            shared::FlowBaseError::KeyExtraction(err @ ValidatorKeyError::MissingValidatorKey) => {
+                rpc_error_response::data(
+                    api_error_code::MISSING_VALIDATOR_KEY,
+                    err.to_string(),
+                    rpc_validator_key_logic::error_data::ValidatorKeyNotAvailable,
+                )
+            }
+            shared::FlowBaseError::KeyExtraction(
+                err @ ValidatorKeyError::ValidatorKeyExtraction,
+            ) => rpc_error_response::simple(
+                api_error_code::VALIDATOR_KEY_EXTRACTION,
+                err.to_string(),
+            ),
+            shared::FlowBaseError::Sign(err) => {
+                rpc_error_response::simple(api_error_code::SIGN, err.to_string())
+            }
+            shared::FlowBaseError::RobonodeClient(
+                ref err @ robonode_client::Error::Call(
+                    robonode_client::AuthenticateError::PersonNotFound(ref scan_result_blob)
+                    | robonode_client::AuthenticateError::FaceScanRejected(ref scan_result_blob)
+                    | robonode_client::AuthenticateError::SignatureInvalid(ref scan_result_blob)
+                    | robonode_client::AuthenticateError::LogicInternal(ref scan_result_blob),
+                ),
+            ) => rpc_error_response::data(
+                api_error_code::ROBONODE,
+                err.to_string(),
+                error_data::ScanResultBlob(scan_result_blob.clone()),
+            ),
+            shared::FlowBaseError::RobonodeClient(
+                err @ robonode_client::Error::Call(
+                    robonode_client::AuthenticateError::FaceScanRejectedNoBlob,
+                ),
+            ) => rpc_error_response::data(
+                api_error_code::ROBONODE,
+                err.to_string(),
+                error_data::ShouldRetry,
+            ),
+            shared::FlowBaseError::RobonodeClient(err) => {
```

Comment:
I think we'd want to unpack the robonode errors explicitly here, instead of going for a wildcard match; I'm talking about enumeration all possible `err @ robonode_client::Error` variants...

Reply:
https://github.com/humanode-network/humanode/pull/1026/commits/4bc57985fe04d89aae419caf9187889d1b0d30ac